include/gatekeeper_lls.h

```diff
@@ -191,6 +196,29 @@ struct lls_config {
 	uint16_t           tx_queue_back;
 };
 
+/*
+ * Functions for resolving IPv4 addresses to link layer addresses.
+ *
+ * resolve_arp() calls the given callback function once when a
+ * resolution is available.
+ *
+ * hold_arp() calls the given callback function function when a
+ * resolution is available and whenever there is an update.
+ *
+ * put_arp() revokes the subscription. If a block calls either
+ * resolve_arp() or hold_arp(), it should later call put_arp()
+ * when it is no longer interested in the map. Callers of
+ * resolve_arp() may choose to put_arp() on a map after some
+ * timeout if no resolution via the callack function is received.
```

Comment:
In the next iteration it will be, because the requesting block should not free the resources until it knows that the `put_arp()` has succeeded. Notice that a side effect of this is that a block could call `hold_arp()`, then `put_arp()`, then receive an update via the callback, then receive notification that the put was successful via a callback. So it's the block's responsibility to make sure that any updates occur before the put is finished will not break the block.